src/Bicep.Core/Registry/IArtifactDispatcher.cs

```diff
@@ -11,13 +11,13 @@
 
 namespace Bicep.Core.Registry
 {
-    public interface IModuleDispatcher : IModuleReferenceFactory
+    public interface IModuleDispatcher : IArtifactReferenceFactory
     {
         RegistryCapabilities GetRegistryCapabilities(ArtifactReference moduleReference);
 
         ArtifactRestoreStatus GetArtifactRestoreStatus(ArtifactReference moduleReference, out DiagnosticBuilder.ErrorBuilderDelegate? errorDetailBuilder);
 
-        ResultWithDiagnostic<Uri> TryGetLocalModuleEntryPointUri(ArtifactReference moduleReference);
+        ResultWithDiagnostic<Uri> TryGetLocalArtifactEntryPointUri(ArtifactReference moduleReference);
```

Comment:
[nit] `moduleReference` -> `artifactReference`

Might be worth searching the codebase for any other occurrences of `ArtifactReference moduleReference`.